cfme/infrastructure/provider/__init__.py

```diff
@@ -184,15 +194,10 @@ def hosts(self):
 
     def get_clusters(self):
         """returns the list of clusters belonging to the provider"""
-        web_clusters = []
-        view = navigate_to(self, 'Details')
-        # todo: create nav location + view later
-        view.contents.relationships.click_at('Clusters')
-        icons = Quadicon.all(qtype='cluster')
+        view = navigate_to(self, 'Clusters')
         cluster_col = self.appliance.get(ClusterCollection)
-        for icon in icons:
-            web_clusters.append(cluster_col.instantiate(icon.name, self))
-        return web_clusters
+        # todo: to handle clusters on multiple pages
```

Comment:
right, good todo here, ```view.entities.get_all()``` should be modified to do this I think

Reply:
get_all(surf_pages=True) already does this. I can't recall why I added this todo instead of adding surf_pages param.